src/stylesheets/components/_site-search.scss

```diff
@@ -242,5 +242,7 @@ $icon-size: 40px;
 .app-site-search__separator {
   display: inline-block;
   margin-right: govuk-spacing(1);
-  margin-left: govuk-spacing(1);
+  // Unfortunate magic number here to account for the non-breaking space created
+  // by visually hidden spans being moved onto new lines in the result markup
+  margin-left: 4px;
```

Comment:
I'm not 100% sure there's a need to change there:
- the `innerHTML` property of the suggestion `<li>` does not show any line breaks or whitespace between the elements (like it does looking at the whole header, for example, where it shows the whitespace in the HTML sent from the server). The inspector shows the `<span>`s on different lines due to it formatting things nicely, but the source of truth for rendering is what's in `innerHTML`.
- the switch to 4px creates a difference in the screeshots of the nodes in this PR and on main, while keeping things to 5px makes the screenshots the same, both on Chrome and Firefox (you can right-click a node in the inspector to take a screenshot, though you'll have to remove the `display: none` on the wrapping `<ul>` to keep the menu visible first)